src/Generator/AbstractTypeGenerator.php

```diff
@@ -62,13 +62,19 @@ function (%s) <closureUseStatements>{
      */
     private $expressionLanguage;
 
+    /**
+     * @var int
+     */
+    protected $cacheDirMask;
+
     /**
      * @param string           $classNamespace The namespace to use for the classes.
      * @param string[]|string  $skeletonDirs
      */
-    public function __construct($classNamespace = self::DEFAULT_CLASS_NAMESPACE, $skeletonDirs = [])
+    public function __construct($classNamespace = self::DEFAULT_CLASS_NAMESPACE, $skeletonDirs = [], $cacheDirMask = 0777)
```

Comment:
This change could introduce some secure issue so we should just give a way to override the value without changing it for this bugfix version, we'll change it in next major version.
```suggestion
    public function __construct($classNamespace = self::DEFAULT_CLASS_NAMESPACE, $skeletonDirs = [], $cacheDirMask = 0775)
```